### PR TITLE
feat(inputs.snmp): Hint to use source tag

### PR DESCRIPTION
**Pull request by Hipska**
# Required for all PRs

- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

Related to #8011 and #4413

Add hint to use the source tag to standardise the snmp input by marking others as deprecated.

### CODE REVIEW

**Comment by srebhan** [comment]
@Hipska while I love the PR, the issue is that you double the cardinality (at least for InfluxDB) users with this change. Can you please add an option to switch between the existing tag and the new `source` tag!

**Comment by Hipska** [comment]
The added tag isn't holding high cardinality, it is one of the values of the `agents` config.

**Comment by powersj** [comment]
While it is not a potential source of lots of different values, it does affects a user's cardinality unnecessarily. Hence the request for a config value to opt-in to the change, rather than force this on users and tell them how to opt-out after they realize the change.

**Comment by Hipska** [comment]
Meaning this PR is useless, as you can opt-in with the `agent_host_tag` setting already. 

The intention of this PR was to have a `source` tag by default and not opt-in.

**Comment by powersj** [comment]
@Hipska what do you want to do with this PR then? Make it a doc change to suggest using "source" or close this?

**Comment by Hipska** [comment]
I still would like to deprecate the `agent_host_tag` setting..

**Comment by powersj** [comment]
ok happy to see that, let us know when the PR is updated and ready for review again please

**Comment by Hipska** [comment]
It is already done..

**Comment by powersj** [comment]
Yes, but you have not changed anything with respect to our previous concerns around a new tag.

**Comment by Hipska** [comment]
Correct.

> The intention of this PR was to have a source tag by default and not opt-in.

**Comment by powersj** [comment]
Thomas,

I am not going to go around with you on this. Both Sven and I are in agreement that we are not making this change. As-is this PR will not be accepted.

If you wish to still deprecate the option, that is fine, but any new option must be opt-in.

**Comment by Hipska** [comment]
Is it okay to have a warning in the log when the value of `agent_host_tag` is not `"source"` or if it is still the default?

**Comment by powersj** [comment]
If you mark the `agent_host_tag` as deprecated, that would result in two warnings for the user, right?

**Comment by Hipska** [comment]
Sorry, I wasn't clear enough. In that case the property wouldn't be deprecated anymore, as we cannot say it is deprecated and at the same time ask to set it to a specific value.

So, I would do an `if(agent_host_tag != "source"){log.warn("..")}` instead of the regular deprecation message. As the current way I did by adding an additional source tag doesn't get accepted.

**Comment by srebhan** [comment]
@Hipska I agree that we need to only conditionally warn if the setting is not `"source"`. So how about
```golang
	if s.AgentHostTag != "source" {
		models.PrintOptionDeprecationNotice(telegraf.Warn, "inputs.snmp", "agent_host_tag", telegraf.DeprecationInfo{
			Since:  "1.29.0",
			Notice: `should be set to "source" and will thus be removed`,
		})
	}
```
which uses the deprecation-style and thus might trigger action while a usual warning will not...

**Comment by Hipska** [comment]
!retry-failed

**Review by srebhan — APPROVED** [review]
Thanks @Hipska!

File: plugins/inputs/snmp/README.md [hunk 1]
```diff
@@ -67,7 +67,8 @@ details.
   ## SNMP community string.
   # community = "public"
 
-  ## Agent host tag
+  ## Agent host tag; should be set to "source"
```
Comment (powersj) [inline plugins/inputs/snmp/README.md:70]: Is it worth saying why we would suggest this?

Something like:

```suggestion
  ## Agent host tag; should be set to "source" for consistent usage across plugins
```

File: plugins/inputs/snmp/snmp.go [hunk 2]
```diff
@@ -108,6 +109,12 @@ func (s *Snmp) Init() error {
 	if len(s.AgentHostTag) == 0 {
 		s.AgentHostTag = "agent_host"
 	}
+	if s.AgentHostTag != "source" {
+		models.PrintOptionValueDeprecationNotice(telegraf.Warn, "inputs.snmp", "agent_host_tag", s.AgentHostTag, telegraf.DeprecationInfo{
+			Since:  "1.29.0",
+			Notice: `should be set to "source"`,
```
Comment (powersj) [inline plugins/inputs/snmp/snmp.go:115]: Same question here.

Comment (Hipska) [reply plugins/inputs/snmp/snmp.go:115]:
```suggestion
			Notice: `should be set to "source" for consistent usage across plugins`,
```

File: plugins/inputs/snmp/sample.conf [hunk 3]
```diff
@@ -30,7 +30,8 @@
   ## SNMP community string.
   # community = "public"
 
-  ## Agent host tag
+  ## Agent host tag; should be set to "source"
```
Comment (Hipska) [inline plugins/inputs/snmp/sample.conf:33]:
```suggestion
  ## Agent host tag; should be set to "source" for consistent usage across plugins
```

**Review by powersj — APPROVED** [review]
Thank you!

**Comment by Hr0bar** [comment]
Think its OK the way it is, but its still incredibly confusing for the user. We got hit by it:

`2024-01-05T06:41:57Z W! DeprecationWarning: Value "agent_host" for option "agent_host_tag" of plugin "inputs.snmp" deprecated since version 1.29.0 and will be removed in : should be set to "source" for consistent usage across plugins`

And had to look at this thread to understand why we are getting a deprecation for a DEFAULT value that we dont explicitly set. But I see it was the default default without setting it explicitly, so hard to find a good decision how to handle... but the docs could exaplain it better perhaps? This is quite confusing in the docs (without looking up more details elsewhere):

```
snmp
tags:
agent_host (deprecated in 1.29: use source instead)
```

It agent_host_tag should be "source" and anything else is deprecated, then perhaps the deprecation should be for the entire agent_host_tag option that should get removed entirely in the future though? Instead of deprecating ANY value other than "source". Otherwise the deprecation warning will be kept forever if the option to set it to something different is kept?

**Comment by srebhan** [comment]
@Hr0bar a PR for an explanation in the README (or config) is welcome!

**Pull request merged**